lib/transactions.rb

```diff
@@ -0,0 +1,23 @@
+require_relative 'income'
+require_relative 'expense'
+
+class Transactions
```

Comment:
I think it's great that there's a useful, low-level object like `Transactions`.  However, it seems a bit far away from the domain of expenses and income.  For example, a user of `Transactions` needs to know to inject `Income` or `Expense`.  Could you add a way to let the user specify their information in a way that's closer to the domain language?